dev/tasks/tasks.yml

```diff
@@ -438,7 +438,7 @@ tasks:
       arch: "arm64"
       arrow_jemalloc: "OFF"
       python_version: "{{ python_version }}"
-      macos_deployment_target: "11.0"
+      macos_deployment_target: "12.0"
```

Comment:
We need to update `big-sur` in `wheel-macos-big-sur-{{ python_tag }}-arm64` too.